### DISCUSSION TITLE
Setting Up A Home Border Router

### DISCUSSION THREAD

**Discussion by Char8603** [openthread/openthread#9425] [Q&A]
Hi everyone. I'm a student who is really interested in exploring everything related to Thread and Matter, and I thought I would work on a personal project that will educate me more on these topics, but I have been running into some issues. I'll give some background.

**Inspiration:** We recently purchased a Schlage Encode Plus Smart Lock that is Thread enabled, however, a border router is required. Rather than buy one, I thought that I would use the tools I have to build my own, following the documentation on openthread.io.

**Components:**

- Raspberry Pi 4 Model B (4GB)
- 2 * Nordic nRF52840 Dongles

**Goal:** Build a boarder router that will connect my current and future thread devices to the internet.

Many of the exercises on openthread.io have given me issue and seem to be targeted for the Nordic nRF52840-DK and not the Dongle. Nevertheless, I was able to successfully flash the RCP firmware to one of the Dongles, and I believe that I have that correct.

At the moment, I am following the documentation found here for building the Border Router: https://openthread.io/guides/border-router/build

Here are the steps I have taken:

1. git clone https://github.com/openthread/ot-br-posix
2. WEB_GUI=1 ./script/bootstrap (after cd into ot-br-posix)
3. INFRA_IF_NAME=wlan0 WEB_GUI=1 ./script/setup
4. Confirmed RCP device is at ttyACM0
5. Restarted Raspberry Pi
6. From here, navigated to https://openthread.io/codelabs/openthread-border-router#1
7. sudo service otbr-agent restart
8. sudo service otbr-agent status (active)
9. sudo ot-ctl dataset init new
10. sudo ot-ctl dataset commit active
11. sudo ot-ctl ifconfig up
12. sudo ot-ctl thread start
13. sudo ot-ctl state (leader)
14. Returned to prior webpage
15. sudo systemctl status
16. mdns.service is active
17. otbr-agent.service is active
18. otbr-web.service is active but with ERROR:
     otbr-web.service - Border Router Web
          Loaded: loaded (/lib/systemd/system/otbr-web.service; enabled; vendor preset: enabled)
          Active: active (running) since Sat 2023-09-16 10:47:52 EDT; 1h 0min ago
      Main PID: 686 (otbr-web)
      Tasks: 1 (limit: 3933)
        CPU: 628ms
     CGroup: /system.slice/otbr-web.service
             └─686 /usr/sbin/otbr-web
     Sep 16 10:47:52 raspberrypi systemd[1]: Started Border Router Web.
     Sep 16 10:47:52 raspberrypi otbr-web[686]: otbr-web[686]: [INFO]-WEB-----: Running 0.3.0-thread-reference-20230>
     Sep 16 10:47:52 raspberrypi otbr-web[686]: otbr-web[686]: [INFO]-WEB-----: Border router web started on wpan0
     Sep 16 10:47:52 raspberrypi otbr-web[686]: otbr-web[686]: [ERR ]-WEB-----: OpenThread daemon is not running.
     Sep 16 10:47:52 raspberrypi otbr-web[686]: [INFO]-WEB-----: Running 0.3.0-thread-reference-20230119-305-gff2f24>
     Sep 16 10:47:52 raspberrypi otbr-web[686]: [INFO]-WEB-----: Border router web started on wpan0
     **Sep 16 10:47:52 raspberrypi otbr-web[686]: [ERR ]-WEB-----: OpenThread daemon is not running.**

I am not too sure where I am going wrong at this point. Navigating to 127.0.0.1:80 displays the OpenThread Border Router home page, and the status section looks meaningful. I can even see the topology report showing that the leader is connected. I'm just not too sure how to proceed from here.

If anyone has any feedback or has gone through a similar situation, I would sincerely appreciate it! Thanks!

**Comment by wgtdkp** [comment]
The next step should be: setting up the smart lock with your Border Router.

0. I believe there is a mobile app for setting up "Schlage Encode Plus Smart Lock" (though I don't know which one).
1. You need to get support from Schlage to make sure that Schlage app is integrated with [GMS Core Thread API](https://developers.home.google.com/reference/com/google/android/gms/threadnetwork/package-summary)
2. You need to add the Thread Network Credentials (aka Active Operational Dataset) of the BR to GMS Core, and then use the Schlage app to set up the lock to add it to the Thread network of the BR
3. You need to write a simple mobile app for step 2 by calling the GMS Core APIs; Google has been working on a codelab for step 2 and it will be publicly available soon.

**Reply by Char8603** [reply]
Thank you for your feedback. May I ask, how will this solve the "OpenThread Daemon is not running" issue?
